src/libraries/Microsoft.Extensions.Http/src/HttpClientFactoryOptions.cs

```diff
@@ -103,5 +109,58 @@ public TimeSpan HandlerLifetime
 
         internal bool SuppressDefaultLogging { get; set; }
         internal List<Action<HttpMessageHandlerBuilder>> LoggingBuilderActions { get; } = new List<Action<HttpMessageHandlerBuilder>>();
+
+        private List<Action<HttpMessageHandlerBuilder>>? _httpMessageHandlerBuilderActions;
+        private List<Action<IPrimaryHandlerBuilder>>? _primaryHandlerActions = new List<Action<IPrimaryHandlerBuilder>>();
+        private List<Action<IAdditionalHandlersBuilder>>? _additionalHandlersActions = new List<Action<IAdditionalHandlersBuilder>>();
+
+        internal IReadOnlyList<Action<IPrimaryHandlerBuilder>> PrimaryHandlerActions => _primaryHandlerActions!;
+        internal IReadOnlyList<Action<IAdditionalHandlersBuilder>> AdditionalHandlersActions => _additionalHandlersActions!;
```

Comment:
Why `IReadOnlyList` rather than `List`? The latter will be more efficient, especially if code ever enumerates these.  And they're internal, so you don't need to worry about breaks if you ever decided to change the type.